src/smolagents/agents.py

```diff
@@ -1209,6 +1220,12 @@ def __init__(
             planning_interval=planning_interval,
             **kwargs,
         )
+        self.stream_outputs = (
+            stream_outputs
+            if stream_outputs is not None
+            else hasattr(self.model, "generate_stream")
+            and has_implemented_method(self.model, Model, "generate_stream")
+        )
```

Comment:
After giving it some thought, I believe that to maintain backward compatibility, streaming should be introduced as an optional feature, and additionally, the default behavior should remain unchanged for users who do not explicitly opt into streaming. This would help ensure a smooth transition without breaking existing workflows.

Reply:
I think this is already resolved.